### DISCUSSION TITLE
How to serialize JSON to std::vector<uint8_t>?

### DISCUSSION THREAD

**Discussion by db4** [nlohmann/json#2548] [Q&A] [answered]
I would like to do the same as `json::dump()` but placing the result into a vector of bytes. What's the recommended way to do that?

**Comment by nlohmann** [comment, marked as answer]
This is currently not possible out of the box. You would need to serialize to a `std::string` and then create a vector out of that:

```cpp
auto str = j.dump();
std::vector<std::uint8_t> data(str.begin(), str.end());
```

**Reply by db4** [reply]
> 
> 
> This is currently not possible out of the box. You would need to serialize to a `std::string` and then create a vector out of that:
> 
> ```c++
> auto str = j.dump();
> std::vector<std::uint8_t> data(str.begin(), str.end());
> ```

So it cannot be done without copying the string to a vector. OK, got it.
